### PR TITLE
refactor: remove prev state in output of setState updater

### PR DESCRIPTION
**Pull request by bolariin**
Resolves https://github.com/opencollective/opencollective/issues/2801

### CODE REVIEW

**Comment by bolariin** [comment]
Looking at this [blog](https://reactjs.org/blog/2017/09/26/react-v16.0.html), it looks like we can return null instead.

**Comment by znarf** [comment]
@bolariin yes, way to go!

File: components/edit-collective/EditGoals.js [hunk 1]
```diff
@@ -100,32 +100,30 @@ class EditGoals extends React.Component {
...
-    this.setState(state => ({ ...state, isTouched: true, goals: [...state.goals, newGoal] }));
+    this.setState(state => ({ isTouched: true, goals: [...state.goals, newGoal] }));
   };
 
   removeGoal = index => {
     this.setState(state => {
-      if (index < 0 || index > state.goals.length) {
-        return state;
-      } else {
+      if (index >= 0 && index <= state.goals.length) {
```
Comment (znarf) [inline components/edit-collective/EditGoals.js]: That is my last feedback, I don't see any reason to change this test, so let's revert it and return null.

File: components/edit-collective/EditMembers.js [hunk 2]
```diff
@@ -158,13 +156,12 @@ class EditMembers extends React.Component {
   removeMember = index => {
     return this.setState(state => {
       const memberEntry = state.members[index];
-      if (memberEntry.member && !this.confirmRemoveMember(memberEntry)) {
-        return state;
-      } else {
+      if (!memberEntry.member || this.confirmRemoveMember(memberEntry)) {
```
Comment (znarf) [inline components/edit-collective/EditMembers.js]: Same as above, I don't see any reason to change the condition, so let's revert and return null.

**Comment by znarf** [comment]
Thank you @bolariin !

**Comment by Betree** [comment]
Hi @bolariin, thanks for your recent contribution! We have a monthly tweet to publicly thank contributors, would you be ok to be mentioned in it? If so, could you share your Twitter handle with me?

**Comment by bolariin** [comment]
@Betree I'm okay with being mentioned but I don't have a twitter handle.

**Pull request merged**